src/script/cpp_api/s_env.cpp

```diff
@@ -128,12 +142,14 @@ void ScriptApiEnv::initializeEnvironment(ServerEnvironment *env)
 				while(lua_next(L, table) != 0){
 					// key at index -2 and value at index -1
 					luaL_checktype(L, -1, LUA_TSTRING);
-					required_neighbors.insert(lua_tostring(L, -1));
+					// Resolve node Ids
+					ndef->getIds(lua_tostring(L, -1), required_neighbors);
 					// removes value, keeps key for next iteration
 					lua_pop(L, 1);
 				}
 			} else if(lua_isstring(L, -1)){
-				required_neighbors.insert(lua_tostring(L, -1));
+				// Resolve node Ids
+				ndef->getIds(lua_tostring(L, -1), required_neighbors);
```

Comment:
Yes it was pointed by est31  that we should resolve nodes after in the process, when Lua finished loading